jobqueue/src/main/java/com/birbit/android/jobqueue/JobManagerThread.java

```diff
@@ -124,7 +134,38 @@ private void handleAddJob(AddJobMessage message) {
             JqLog.e(t, "job's onAdded did throw an exception, ignoring...");
         }
         callbackManager.notifyOnAdded(jobHolder.getJob());
-        consumerManager.onJobAdded();
+        if (insert) {
+            consumerManager.onJobAdded();
+            if (oldJob != null) { //the job was running, will be cancelled if it fails
```

Comment:
I think there is a race condition here. If the app crashes before here, we'll have two jobs with the same singleId. Is ugly but i guess we'll have to make it a transaction in the persistentQueue such that cancellation of the old one (if exists) and the new job should be done together. (haven't checked below yet so ignore if you are already handling it).